java/src/main/java/utility/XClientTraceIdClientInterceptor.java

```diff
@@ -0,0 +1,35 @@
+package utility;
+
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.grpc.*;
+
+public class XClientTraceIdClientInterceptor implements ClientInterceptor {
+    private static final Logger logger = LoggerFactory.getLogger(XClientTraceIdClientInterceptor.class.getClass());
+    private static final Metadata.Key<String> X_CLIENT_TRACE_ID = Metadata.Key.of("x-client-trace-id", Metadata.ASCII_STRING_MARSHALLER);
+
+    @Override
+    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method,
+                                                               CallOptions callOptions, Channel next) {
+        return new ForwardingClientCall.SimpleForwardingClientCall<>(next.newCall(method, callOptions)) {
+
+            @Override
+            public void start(Listener<RespT> responseListener, Metadata headers) {
+                String xClientTraceId = UUID.randomUUID().toString();
+                headers.put(X_CLIENT_TRACE_ID, xClientTraceId);
+                logger.info("sending request for xClientTraceId {}", xClientTraceId);
+
+                super.start(new ForwardingClientCallListener.SimpleForwardingClientCallListener<>(responseListener) {
+                    @Override
+                    public void onClose(Status status, Metadata trailers) {
+                        logger.info("request completed for xClientTraceId {} with status {}", xClientTraceId, status);
```

Comment:
This differs slightly from the Java example (linked above), but the intention here is to basically just log the `x-client-trace-id` that was passed any time a request completes. This log line should be hit whether the request completes successfully or with an error. I'll include some sample output below

example log for a successful call:
```
request completed for xClientTraceId 8895e3a5-65d9-4428-bc1e-6e991838f164 with status Status{code=OK, description=null, cause=null}
```

example log for a failed call:
```
request completed for xClientTraceId eb62166d-5126-4b52-857b-ed73c2d3018e with status Status{code=UNAVAILABLE, description=RST_STREAM closed stream. HTTP/2 error code: NO_ERROR, cause=null}
```

Reply:
nice find. Curious, how did you generate the RST_STREAM error?

Reply:
I used the Subscribe example to open a subscription then killed the backend server that was serving the subscription